crates/oxc_linter/src/rules/react/jsx_no_target_blank.rs

```diff
@@ -0,0 +1,660 @@
+    #[diagnostic(severity(warning), help("add rel=`noreferrer` or rel=`noopener` to the element"))]
+    NoTargetBlankWithoutNoopener(#[label] Span),
+}
+
+#[derive(Debug, Clone)]
+pub struct JsxNoTargetBlank {
+    enforce_dynamic_links: EnforceDynamicLinksEnum,
+    warn_on_spread_attributes: bool,
+    allow_referrer: bool,
+    links: bool,
+    forms: bool,
+}
+
+#[derive(Debug, Clone)]
+enum EnforceDynamicLinksEnum {
+    Always,
+    Never,
+}
+
+impl Default for JsxNoTargetBlank {
+    fn default() -> Self {
+        Self {
+            enforce_dynamic_links: EnforceDynamicLinksEnum::Always,
+            warn_on_spread_attributes: false,
+            allow_referrer: false,
+            links: true,
+            forms: false,
+        }
+    }
+}
+
+impl JsxNoTargetBlank {
+    fn diagnostic(&self, span: Span, ctx: &LintContext) {
+        if self.allow_referrer {
+            ctx.diagnostic(JsxNoTargetBlankDiagnostic::NoTargetBlankWithoutNoopener(span));
+        } else {
+            ctx.diagnostic(JsxNoTargetBlankDiagnostic::NoTargetBlankWithoutNoreferrer(span));
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule aims to prevent user generated link hrefs and form actions from creating security vulnerabilities by
+    /// requiring rel='noreferrer' for external link hrefs and form actions, and optionally any dynamically generated link
+    /// hrefs and form actions.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// When creating a JSX element that has an a tag, it is often desired to have the link open in a new tab using the
+    /// target='_blank' attribute. Using this attribute unaccompanied by rel='noreferrer', however, is a severe security
+    /// vulnerability (see noreferrer docs and noopener docs for more details) This rules requires that you accompany
+    /// target='_blank' attributes with rel='noreferrer'.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// /// correct
+    /// var Hello = <p target="_blank"></p>
+    /// var Hello = <a target="_blank" rel="noreferrer" href="https://example.com"></a>
+    /// var Hello = <a target="_blank" rel="noopener noreferrer" href="https://example.com"></a>
+    /// var Hello = <a target="_blank" href="relative/path/in/the/host"></a>
+    /// var Hello = <a target="_blank" href="/absolute/path/in/the/host"></a>
+    /// var Hello = <a></a>
+    /// /// incorrect
+    /// var Hello = <a target='_blank' href="https://example.com/"></a>
+    /// var Hello = <a target='_blank' href={dynamicLink}></a>
+    /// ```
+    JsxNoTargetBlank,
+    correctness
+);
+
+impl Rule for JsxNoTargetBlank {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::JSXOpeningElement(jsx_ele) = node.kind() {
+            // TODO need to read from configuration: `settings: { linkComponents: ['Link'] }`
```

Comment:
use `get_element_type` or is that just for jsx_a11y rules?

Reply:
yes, I think get_element_type is just for jsx_a11y rules，the rule setting is like this:
![image](https://github.com/oxc-project/oxc/assets/42141272/fb4cfa62-b278-4e0b-9a5c-01279cdab1ad)